scheduler/core/eventsqueue/events.py

```diff
@@ -73,34 +73,21 @@ class WeatherChange(Interruption):
     new_conditions: Conditions
 
 
-@dataclass
-class Blockage(Event, ABC):
-    """
-    Parent class for any interruption that causes a blockage and requires a resume event.
-    """
-    end: Optional[datetime] = None  # needs a resume event
-
-    def ends(self, end: datetime) -> None:
-        self.end = end
-
-    def time_loss(self) -> timedelta:
-        if self.end:
-            return self.end - self.start
-        else:
-            raise ValueError("Can't calculate Blockage time loss without end value.")
-
-
-@dataclass
-class ResumeNight(Interruption):
-    """
-    Event that lets the scheduler knows that the night can be resumed.
-    """
-    ...
-
-
 @final
-class Fault(Blockage):
+@dataclass(unsafe_hash=True)
+class Fault(Interruption):
     """
     Blockage that occurs when one or more resources experience a fault.
     """
+    id: str
+    end: datetime
+    time_loss: timedelta
```

Comment:
This should probably be divided into two events:

1. `Fault`
2. `FaultResolved`

This would correspond to what we will see in Operations mode.

In Validation mode, we know from the data in the files:
1. When a fault occurs
2. When a fault resolves

but I think we are trying to model two concepts here into one thing, which is going to create significant technical debt for later. We could instead model each fault entry in a Validation mode file into both a `Fault` and a `FaultResolved` event linked by ID.

* The `Fault.start` would be the start time of the fault.
* The `FaultResolved.start` would be the start time at which the fault was resolved.

Then we can get rid of:
* `end`
* `timeloss`

and `timeloss` can just be calculated by:

```python
fault_resolved.start - fault.start
```

It may be misleading to call a `FaultResolved` an `Interruption`, so we might want to tweak the event hierarchy slightly, but I don't think we need to invest a lot of time there.